x-pack/plugin/async-search/qa/rest/build.gradle

```diff
@@ -10,6 +11,10 @@ esplugin {
   classname 'org.elasticsearch.query.DeprecatedQueryPlugin'
 }
 
+dependencies {
+  clusterPlugins project(xpackModule('async-search'))
```

Comment:
FWIW, this _should_ be unnecessary as we automatically add the current project as a dependency here. Essentially, no need to say "I depend on myself".

Reply:
Without it, the test cluster orchestration throws an `Unable to locate plugin 'x-pack-async-search'` exception.
I had quite the headache around dependencies, and this was the only way I found. There may be a better way or a hidden problem somewhere, but I didn't find it 👀 

A sibling project `qa:security` uses `javaRestTestImplementation project(xpackModule('async-search'))`, but it didn't work here either.

Reply:
Ah, that's a bug likely related to the use of `base-internal-es-plugin`. I'll get that fixed. Having this here is fine for now though.